src/sage/rings/polynomial/integer_valued_polynomials.py

```diff
@@ -0,0 +1,1125 @@
+
+        class Element(CombinatorialFreeModule.Element):
+
+            def umbra(self):
+                """
+                Return the Bernoulli umbra.
+
+                This is the derivative at `-1` of the shift by one.
+
+                EXAMPLES::
+
+                    sage: F = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: B = F.gen()
+                    sage: (B+1).umbra()
+                    3/2
+
+                TESTS::
+
+                    sage: [(B**n).umbra() for n in range(1, 11)]
+                    [1/2, 1/6, 0, -1/30, 0, 1/42, 0, -1/30, 0, 5/66]
+                """
+                return self.shift().derivative_at_minus_one()
+
+            def delta(self):
+                r"""
+                Return the image by the difference operator `\Delta`.
+
+                The operator `\Delta` is defined on polynomials by
+
+                .. MATH::
+
+                    f \mapsto f(x+1)-f(x).
+
+                EXAMPLES::
+
+                    sage: F = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: S = F.basis()
+                    sage: S[5].delta()
+                    S[0] + S[1] + S[2] + S[3] + S[4]
+                """
+                return self.variable_shift() - self
+
+            def variable_shift(self, k=1):
+                r"""
+                Return the image by the shift of variables.
+
+                On polynomials, the action is the shift
+                on variables `x \mapsto x + 1`.
+
+                INPUT:
+
+                - `k` -- integer (default: 1)
+
+                EXAMPLES::
+
+                    sage: A = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: S = A.basis()
+                    sage: S[5].variable_shift()
+                    S[0] + S[1] + S[2] + S[3] + S[4] + S[5]
+                """
+                if k == 0:
+                    return self
+
+                A = self.parent()
+                resu = A.sum(c * A.monomial(j) for i, c in self
+                             for j in range(i + 1))
+                if k == 1:
+                    return resu
+                return resu.variable_shift(k - 1)
+
+            def variable_unshift(self, k=1):
+                r"""
+                Return the image by the unshift of variables.
+
+                On polynomials, the action is the shift
+                on variables `x \mapsto x - k`.
+
+                INPUT:
+
+                - `k` -- integer (default: 1)
+
+                EXAMPLES::
+
+                    sage: A = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: S = A.basis()
+                    sage: S[5].variable_unshift()
+                    -S[4] + S[5]
+                """
+                if k == 0:
+                    return self
+
+                A = self.parent()
+                resu = (A.sum(c * A.monomial(i) for i, c in self) -
+                        A.sum(c * A.monomial(i - 1) for i, c in self if i))
```

Comment:
`A.linear_combination()` or `A._from_dict` for speed. Although isn't the first summation just `self`?